classes/driver/field/input/number.php

```diff
@@ -20,6 +20,22 @@ public function checkValidation($inputValue, $formData = null)
     }
 
     /**
+     * checks requirement state
+     *
+     * @param $inputValue
+     * @param null $formData
+     * @return bool
+     */
+    public function checkRequirement($inputValue, $formData = null)
+    {
+        if (!$this->isMandatory()) {
+            return true;
+        }
+
+        return is_int($inputValue) && !empty($inputValue);
```

Comment:
Will return false if "0" is sent because of `!empty($inputValue)`.

Also $inputValue is already casted in (int) before.